app/IATI/Traits/MigrateActivityPublishedTrait.php

```diff
@@ -0,0 +1,483 @@
+
+                return "{$publisherId}-activities.xml";
+            }
+
+            return $activityPublished->first()->filename;
+        }
+
+        return null;
+    }
+
+    /**
+     * Migrate merged file.
+     *
+     * @param $aidStreamOrganization
+     * @param $iatiOrganization
+     *
+     * @return object|null
+     *
+     * @throws \DOMException
+     */
+    public function migrateActivityMergedFile($aidStreamOrganization, $iatiOrganization): ?object
+    {
+        $activityPublished = $this->getIatiActivityPublished($iatiOrganization);
+        $publishedFiles = $activityPublished ? $activityPublished->published_activities : [];
+
+        if ($publishedFiles) {
+            $aidstreamActivityPublished = $this->db::connection('aidstream')->table('activity_published')
+                                                   ->where('organization_id', $aidStreamOrganization->id)->where(
+                                                       'published_to_register',
+                                                       1
+                                                   )->get();
+            $aidstreamMergedFilePath = 'aidstream-xml';
+            $aidstreamMergedFilename = $this->getAidStreamMergedFilename(
+                $aidStreamOrganization,
+                $aidstreamActivityPublished
+            );
+
+            $iatiMergedFilename = "{$iatiOrganization->publisher_id}-activities.xml";
+            $iatiMergedFile = "xml/mergedActivityXml/{$iatiMergedFilename}";
+
+            $this->logInfo("Started migration of merged file for Aidstream org: {$aidStreamOrganization->id}.");
+            $existingContents = awsGetFile("{$aidstreamMergedFilePath}/{$aidstreamMergedFilename}");
+
+            if ($existingContents) {
+                $existingContents = $this->replaceDocumentLinkInXml($existingContents, $iatiOrganization->id);
+                awsUploadFile($iatiMergedFile, $existingContents);
+            } else {
+                $this->xmlGenerator->getMergeXml($publishedFiles, $iatiMergedFilename);
+                $mergedContents = awsGetFile($iatiMergedFile);
+                $mergedContents = $this->replaceDocumentLinkInXml($mergedContents, $iatiOrganization->id);
+                awsUploadFile($iatiMergedFile, $mergedContents);
+            }
+
+            $this->logInfo("Completed migration of merged file for Aidstream org: {$aidStreamOrganization->id}.");
+        } else {
+            $message = 'No activity file to merge.';
+            $this->logInfo($message);
+        }
+
+        return $activityPublished;
+    }
+
+    /**
+     * Returns the latest activity_published of an organization.
+     *
+     * @param $iatiOrganization
+     *
+     * @return ?object
+     */
+    public function getIatiActivityPublished($iatiOrganization): ?object
+    {
+        return (new ActivityPublished())->where('organization_id', $iatiOrganization->id)?->first();
```

Comment:
if ActivityPublished used in multiple places, initialize in constructor. If not, then this is ok for now.